GMG/GMG/UI/Components/Waveform/WaveformExtractor.swift

```diff
@@ -0,0 +1,164 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import Foundation
+import AVFoundation
+
+enum WaveformError: Error {
+    case noAudioTrack
+    case readerFailed(String)
+}
+
+enum DownsampleMode { case rms, peak }
+
+struct WaveformExtractor {
+    
+    //MARK: 디코딩 파이프가 정상 작동하는 지 샘플 개수 세어서 확인하는 함수 -> 동작에는 크게 필요없지만 debug용으로 작동하는 함수
+    static func debugCountSamples(from url: URL, targetSampleRate: Double = 44_100) async throws -> Int {
+        let asset = AVURLAsset(url: url)
+        
+        let audioTracks = try await asset.loadTracks(withMediaType: .audio)
+        guard let track = audioTracks.first else {
+            throw WaveformError.noAudioTrack
+        }
+        
+        let outputSettings: [String: Any] = [
+            AVFormatIDKey: kAudioFormatLinearPCM,
+            AVLinearPCMIsFloatKey: true,
+            AVLinearPCMIsBigEndianKey: false,
+            AVLinearPCMBitDepthKey: 32,
+            AVNumberOfChannelsKey: 1,
+            AVSampleRateKey: targetSampleRate
+        ]
+        
+        let reader = try AVAssetReader(asset: asset)
+        let output = AVAssetReaderTrackOutput(track: track, outputSettings: outputSettings)
```

Comment:
재사용되는 변수가 아니면, 따로 선언하지 않는게 가독성이 더 좋을 것 같아요

```suggestion
        let reader = try AVAssetReader(asset: asset)
        let output = AVAssetReaderTrackOutput(track: track, outputSettings: [
            AVFormatIDKey: kAudioFormatLinearPCM,
            AVLinearPCMIsFloatKey: true,
            AVLinearPCMIsBigEndianKey: false,
            AVLinearPCMBitDepthKey: 32,
            AVNumberOfChannelsKey: 1,
            AVSampleRateKey: targetSampleRate
        ])
```